### PR TITLE
OfficeFloor implementation of Framework Benchmark Performance Tests.

### PR DESCRIPTION
**Pull request by sagenschneider**

### CODE REVIEW

**Comment by blee-techempower** [comment]
Passed local test.
```
 Test: officefloor
|       fortune     : PASS
|       plaintext   : PASS
|       db          : PASS
|       update      : PASS
|       json        : PASS
|       query       : PASS

```

**Comment by msmith-techempower** [comment]
@sagenschneider I had to revert your pull request as it did not conform to our standards. I will leave comments on the files needing changes here, but you will have to open a new pull request to get OfficeFloor added.

File: frameworks/Java/officefloor/setup.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+echo "Loading maven ..."
+sudo add-apt-repository "deb http://ppa.launchpad.net/natecarlson/maven3/ubuntu precise main"
+sudo apt-get update
+sudo apt-get -y --force-yes install maven3
+if [ -e /usr/bin/mvn ]
+then
+    sudo rm -f /usr/bin/mvn
+fi
+sudo ln -s /usr/share/maven3/bin/mvn /usr/bin/mvn
+
```
Comment (msmith-techempower) [inline frameworks/Java/officefloor/setup.sh]: Please delete everything about this and replace with:
```bash
fw_depends java mvn
```

We are presently in the midst of opening a pull request to fix up the various java version problems (likely, the culprit behind why you did your dependencies this way), so once we get that in (hopefully, today or tomorrow) you can open your new pull request.

**Comment by sagenschneider** [comment]
No worries.  Please excuse this if it has caused too many problems.  I'm happy to make the change once the issues are resolved.  Please either update this PR once done, or email me at daniel@officefloor.net  and I'll resubmit with changes.

Also, can I confirm that the fix will have:
  - "fw_depends java8" use the java8 libraries.  Current implementation seemed to replace the libraries with Java 7 versions.
 - "fw_depends mvn" will pull in maven 3.1.1 (or possibly higher).   There is a bug in mvn 3.0.5 regarding depending on project source dependences causing the OfficeFloor plugin to not start the test application (3.0.5 tries to unzip the source directory).

Note: the above is fixes if that helps.

**Comment by msmith-techempower** [comment]
Really, the only problem was the manual installation of java and maven; we provide `fw_depends` to ensure the environment is built in a way that is easily cleaned.

Additionally, the java8-vs-java7 issues should be resolved by a pull request soon, and I will ping you here when you can rebase and open another pull request.

**Comment by sagenschneider** [comment]
Would you accept a pull request change to toolset/setup/linux/systools/maven.sh setup that is following?  (this will load latest maven)

#!/bin/bash

RETCODE=$(fw_exists ${IROOT}/maven.installed)
[ ! "$RETCODE" == 0 ] || { \
  source $IROOT/maven.installed
  return 0; }

sudo add-apt-repository "deb http://ppa.launchpad.net/natecarlson/maven3/ubuntu precise main"
sudo apt-get update
sudo apt-get -y --force-yes install maven3
if [ -e /usr/bin/mvn ]
then
    sudo rm -f /usr/bin/mvn
fi
sudo ln -s /usr/share/maven3/bin/mvn /usr/bin/mvn

mvn -version

touch $IROOT/maven.installed

source $IROOT/maven.installed

**Comment by msmith-techempower** [comment]
Why? Does OfficeFloor require a particular version or maven?

**Comment by sagenschneider** [comment]
I am running OfficeFloor through the OfficeFloor maven plugins.  The OfficeFloor plugins load all necessary dependencies, however due to a bug in 3.0.5 the maven dependency loading fails on the source directory when pulling in (as assumes everything to be jars).  This has been resolved in later maven version.  Note: maven 3.0.5 is nearly 3 years old.

**Comment by sagenschneider** [comment]
PR  #1911 has the request changes for OfficeFloor to use "fw_depends java mvn".
Also, PR #1912 has the changes for using the latest version of maven.

**Pull request merged**